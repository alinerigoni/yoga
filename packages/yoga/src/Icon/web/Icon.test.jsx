import React from 'react';
import { render } from '@testing-library/react';

import { ThemeProvider, Icon } from '../..';

describe('Snapshots', () => {
  it('should match snapshot', () => {
    const Circle = props => (
      <svg {...props}>
        <circle cx="16" cy="16" r="16" />
      </svg>
    );
    const { container } = render(
      <ThemeProvider>
        <Icon as={Circle} width="small" height="small" fill="stamina" />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
