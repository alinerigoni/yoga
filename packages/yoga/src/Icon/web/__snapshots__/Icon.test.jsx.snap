// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots should match snapshot 1`] = `
.c0 {
  width: 16px;
  height: 16px;
}

<div>
  <svg
    class="c0"
    fill="#231B22"
    height="16"
    width="16"
  >
    <circle
      cx="16"
      cy="16"
      r="16"
    />
  </svg>
</div>
`;

exports[`Snapshots should match snapshot with size prop 1`] = `
.c0 {
  width: 16px;
  height: 16px;
}

<div>
  <svg
    class="c0"
    fill="#231B22"
    height="16"
    width="16"
  >
    <circle
      cx="16"
      cy="16"
      r="16"
    />
  </svg>
</div>
`;

exports[`Snapshots should match snapshot with system props 1`] = `
.c0 {
  box-shadow: 0 5px 5px -3px rgba(152,152,166,0.2),0 8px 10px 1px rgba(152,152,166,0.14),0 3px 14px 2px rgba(152,152,166,0.12);
  margin: 20px;
  width: 16px;
  height: 16px;
}

<div>
  <svg
    class="c0"
    elevation="medium"
    fill="#231B22"
    height="16"
    margin="medium"
    width="16"
  >
    <circle
      cx="16"
      cy="16"
      r="16"
    />
  </svg>
</div>
`;
